### ISSUE TITLE
Allow for interruptions in C functions

### ISSUE THREAD

**Issue by zkamvar** [grunwaldlab/poppr#86]
If somethings running too long, the user should be allowed to stop it. 

from: https://cran.r-project.org/doc/manuals/r-release/R-exts.html#Allowing-interrupts

``` c
#include <R_ext/Utils.h>

void R_CheckUserInterrupt(void);
```

**Comment by zkamvar** [comment]
Note, I found example usage in Karl Broman's pacakge: https://github.com/kbroman/broman/commit/39c297c771f41d2bbd9787bc4aac414db9919e1f

I tried placing this in the inner loop for bruvo's distance and the computation on the microbov data set went from ~9s to ~14s. For some reason, the user time quadrupled, so I simply placed it in the outer loop.

One thing I also noticed about this was that the request to interrupt lagged a bit without the code, but was instantaneous with the code, so there's some benefit in there.
